app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -296,7 +291,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
     protected void onResume() {
         super.onResume();
 
-        updateLaunchButtonText(Pokefly.isRunning());
+        LocalBroadcastManager.getInstance(this).sendBroadcast(new Intent(Pokefly.ACTION_UPDATE_UI));
```

Comment:
Ah sorry I get it: the point is that this intent triggers an UI update, but delays it: `onResume` can be called after we send the intent to start Pokefly, the new intent is delivered *after* the intent to start Pokefly (because intents are delivered in order). The ordering is not really documented but likely enough to work: http://stackoverflow.com/a/28513424/53974. I'll add a comment after merging this.